hw1/src/Triangle.ts

```diff
@@ -0,0 +1,31 @@
+import { Point } from './point';
+import { Shape } from './Shape';
+
+export class Triangle extends Shape {
+  constructor(points: Point[], color?: string, filled?: boolean);
+  constructor(point1: Point, point2: Point, point3: Point);
+
+  constructor(point1: unknown, point2: unknown, point3: unknown) {
+    if(point1 instanceof Point && point2 instanceof Point && point3 instanceof Point) {
+      super([point1, point2, point3]);
+      this.points = [point1, point2, point3];
+    }
+  }
+
+  public toString() {
+    const str = this.points.reduce((a, c, i) => a + `v${i + 1}=${c.toString()},`, '');
+    
+    return `Triangle[${str.slice(0, str.length - 1)}]`
+  }
+
+  public getType() {
+    const res = this.points
+      .map((e, i, arr) => +e.distance(arr[(i+1) % arr.length]).toFixed(1))
```

Comment:
do we really need all these maps, `these.points` is always only three objects, maybe there is a way to write it easier?

Reply:
We can do this just with conditions but I think 1 filter is not so bad ?)